drivers/ethernet/eth_renesas_ra.c

```diff
@@ -0,0 +1,415 @@
+		ctx->ctrl.previous_link_status = ETHER_PREVIOUS_LINK_STATUS_DOWN;
+		ctx->ctrl.link_establish_status = ETHER_LINK_ESTABLISH_STATUS_DOWN;
+		net_eth_carrier_off(ctx->iface);
+	}
+}
+
+static void renesas_ra_eth_initialize(struct net_if *iface)
+{
+	fsp_err_t err;
+	const struct device *dev = net_if_get_device(iface);
+	struct renesas_ra_eth_context *ctx = dev->data;
+	const struct renesas_ra_eth_config *cfg = dev->config;
+
+	LOG_DBG("eth_initialize");
+
+	net_if_set_link_addr(iface, ctx->mac, sizeof(ctx->mac), NET_LINK_ETHERNET);
+
+	if (ctx->iface == NULL) {
+		ctx->iface = iface;
+	}
+
+	ethernet_init(iface);
+
+	const char *phy_connection_type = DT_INST_PROP_OR(0, phy_connection_type, "rmii");
+
+	R_PMISC->PFENET = (uint8_t)(0x0 << R_PMISC_PFENET_PHYMODE0_Pos);
+	/* Handle for phy-connection-type is different from default (rmmi) */
+	if (strcmp(phy_connection_type, "rmii") != 0) {
+		if (strcmp(phy_connection_type, "mii") == 0) {
+			/* Configure pins for MII or RMII. Set PHYMODE0 if MII is selected. */
+			R_PMISC->PFENET = (uint8_t)(0x1 << R_PMISC_PFENET_PHYMODE0_Pos);
+		} else {
+			LOG_ERR("Failed to init ether - phy-connection-type not support");
+		}
+	}
+
+	err = R_ETHER_Open(&ctx->ctrl, cfg->p_cfg);
+
+	if (err != FSP_SUCCESS) {
+		LOG_ERR("Failed to init ether - R_ETHER_Open fail");
+	}
+
+	err = R_ETHER_CallbackSet(&ctx->ctrl, renesas_ra_eth_callback, dev, NULL);
+
+	if (err != FSP_SUCCESS) {
+		LOG_ERR("Failed to init ether - R_ETHER_CallbackSet fail");
+	}
+
+	phy_link_callback_set(cfg->phy_dev, &phy_link_state_changed, (void *)dev);
+	/* Do not start the interface until PHY link is up */
+	net_if_carrier_off(ctx->iface);
+}
+
+static int renesas_ra_eth_tx(const struct device *dev, struct net_pkt *pkt)
+{
+	fsp_err_t err = FSP_SUCCESS;
+	struct renesas_ra_eth_context *ctx = dev->data;
+	uint16_t len = net_pkt_get_len(pkt);
+	static uint8_t tx_buf[NET_ETH_MAX_FRAME_SIZE] __aligned(4);
+
+	if (net_pkt_read(pkt, tx_buf, len)) {
+		goto error;
+	}
+
+	/* Check if packet length is less than minimum Ethernet frame size */
+	if (len < NET_ETH_MINIMAL_FRAME_SIZE) {
+		/* Add padding to meet the minimum frame size */
+		memset(tx_buf + len, 0, NET_ETH_MINIMAL_FRAME_SIZE - len);
+		len = NET_ETH_MINIMAL_FRAME_SIZE;
+	}
+
+	err = R_ETHER_Write(&ctx->ctrl, tx_buf, len);
+	if (err != FSP_SUCCESS) {
+		goto error;
+	}
+
+	return 0;
+
+error:
+	LOG_ERR("Writing to FIFO failed");
+	return -1;
+}
+
+static const struct ethernet_api api_funcs = {
+	.iface_api.init = renesas_ra_eth_initialize,
+	.get_capabilities = renesas_ra_eth_get_capabilities,
+	.send = renesas_ra_eth_tx,
+};
+
+static void renesas_ra_eth_isr(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	ether_eint_isr();
+}
+
+static struct net_pkt *renesas_ra_eth_rx(const struct device *dev)
+{
+	fsp_err_t err = FSP_SUCCESS;
+	struct renesas_ra_eth_context *ctx;
+	struct net_pkt *pkt = NULL;
+	uint32_t len = 0;
+	static uint8_t rx_buf[NET_ETH_MAX_FRAME_SIZE] __aligned(4);
```

Comment:
Is the aligned really needed here, I mean the variables should be aligned properly by the compiler anyway. Of course this works also, but looks unnecessary.

Reply:
It's removed, thank you